webtau-http-groovy/src/test/groovy/com/twosigma/webtau/http/HttpGroovyTest.groovy

```diff
@@ -468,6 +471,72 @@ class HttpGroovyTest implements HttpConfiguration {
         assertFalse(Files.exists(requestFile))
     }
 
+    @Test
+    void "appropriate capture files are generated"() {
+        def requestBody = [a: 'b', c: 1]
+        def requestHeaders = [testheader: 'testValue', another: 'value']
+        http.post("echo-body-and-header", http.header(requestHeaders), requestBody) {
+            body.a.should == 'b'
+            header.testheader.should == 'testValue'
+            header.another.should == 'value'
+        }
+
+        String artifactName = 'echo-body-and-header'
+        http.doc.capture(artifactName)
+
+        Path docRoot = DocumentationArtifactsLocation.resolve(artifactName)
+
+        Path requestBodyFile = docRoot.resolve("request.json")
+        assertTrue(Files.exists(requestBodyFile))
+        def capturedRequestBody = JsonUtils.deserializeAsMap(FileUtils.fileTextContent(requestBodyFile))
+        capturedRequestBody.should == requestBody
+
+        Path responseBodyFile = docRoot.resolve("response.json")
+        assertTrue(Files.exists(responseBodyFile))
+        def capturedResponseBody = JsonUtils.deserializeAsMap(FileUtils.fileTextContent(responseBodyFile))
+        capturedResponseBody.should == requestBody
```

Comment:
seems like a bit of duplication

Reply:
Yeah, there's enough duplication to make it look duplicative but not enough that things look much simpler with it removed.  I've pushed a change to reduce it a bit, let me know what you think?